src/blant.c

```diff
@@ -575,34 +587,74 @@ int RunBlantInThreads(int k, int numSamples, GRAPH *G)
         for(i=0; i<_numCanon; i++) for(j=0; j<_numCanon; j++) _graphletDistributionTable[i][j] = 0;
     }
 
-    if(_THREADS == 1)
-	return RunBlantFromGraph(k, numSamples, G);
+    int indexSamplingRange[2];
+    if(_THREADS == 1) {
+        if (_sampleMethod == SAMPLE_INDEX) {
+            indexSamplingRange[0] = 0;
+            indexSamplingRange[1] = G->n - 1;
+        }
+        return RunBlantFromGraph(k, numSamples, G, _sampleMethod == SAMPLE_INDEX ? indexSamplingRange : NULL);
```

Comment:
I don't understand the point of this indexSamplingRange. Why was it introduced? It seems an entirely un-necessary extra parameter. I think if THREADS==1, then we just ask for numSamples and the existing code is correct, no changes necessary; if THEADS>1, then numSamples = -1 for all child processes (and the child processes should STILL remove duplicates of their own before passing down the pipe to the parent), and then the parent will do a second stage of duplicate removal--we don't want the parent process removing *all* duplicates, because with enough parallelism the parent will have too many duplicates to remove and cause a traffic jam, slowing the children down.